cmd/build.go

```diff
@@ -179,6 +178,32 @@ func subcommandBuild(targetOS string, packagingTask packaging.Task) {
 	assertHoverInitialized()
 	packagingTask.AssertInitialized()
 
+	if buildGoFlutterBranch == config.BuildBranchDefault && config.GetConfig().Branch != "" {
+		buildGoFlutterBranch = config.GetConfig().Branch
+	}
+	if buildCachePath == "" && config.GetConfig().CachePath != "" {
+		buildCachePath = config.GetConfig().CachePath
+	}
+	if buildEngineVersion == config.BuildEngineDefault && config.GetConfig().Engine != "" {
+		log.Warnf("changing the engine version can lead to undesirable behavior")
+		buildEngineVersion = config.GetConfig().Engine
+	}
+	if buildOpenGlVersion == config.BuildOpenGlVersionDefault && config.GetConfig().OpenGL != "" {
+		buildOpenGlVersion = config.GetConfig().OpenGL
+	}
+	if buildVersionNumber == "" {
+		buildVersionNumber = pubspec.GetPubSpec().Version
+	}
+
+	if buildCachePath == "" {
+		buildCachePath = enginecache.DefaultCachePath()
+	}
+	if buildSkipEngineDownload {
+		engineCachePath = enginecache.EngineCachePath(targetOS, buildCachePath)
+	} else {
+		engineCachePath = enginecache.ValidateOrUpdateEngine(targetOS, buildEngineVersion)
```

Comment:
I think this move breaks `hover run`...

Reply:
Yes, `hover run` depends on these values being correct as well.. Both the `hover run` and `hover build` subcommand call `buildGoBinary`, that's why the `buildGo*` variables are set there.